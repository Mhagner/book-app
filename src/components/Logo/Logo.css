.logo__container {
    display: flex;
    align-items: center;  
    margin-bottom: 4.4rem;  
}

.logo__image {
    max-width: 10.44rem;
    margin-right: 1.66rem;
}

.logo__text {
    font-weight: 300;
    font-size: 2.8rem;
    color: var(--grey-dark-1);
}

.logo__text--light {
    color: var(--color-white);
}