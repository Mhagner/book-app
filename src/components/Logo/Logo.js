import './Logo.css'
import LogoIoLight from '../../assets/images/logo.png'
import LogoIoDark from '../../assets/images/logo-dark.png'

const Logo = ({ color = "light" }) => {
    return (
        <div data-testid="logo" className="logo__container">
            {color === "light" ?
                <img className="logo__image" src={LogoIoLight} alt="ioasys" />
                :
                <img className="logo__image" src={LogoIoDark} alt="ioasys" />
            }
            <span
                className={`logo__text ${(color === "light") ? ' logo__text--light' : ''}`}>
                Books
            </span>
        </div>
    )
}

export default Logo