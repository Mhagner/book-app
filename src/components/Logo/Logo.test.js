import { render } from '@testing-library/react'

import Logo from './Logo'

describe('Logo component', () => {
    const renderComponent = (props) => {
        return render(<Logo {...props}/>)
    }

    it('Componente should defined', () => {
        const { getByTestId } = renderComponent()

        const element = getByTestId('logo')

        expect(element).toBeDefined()
    })

    it('Componente should render logo light default', () => {

        const { getByTestId } = renderComponent()

        const element = getByTestId('logo')

        expect(element).toMatchSnapshot()
    })

    it('Componente should render logo dark', () => {
        const props = {
            color: 'dark'
        }

        const { getByTestId } = renderComponent(props)

        const element = getByTestId('logo')

        expect(element).toMatchSnapshot()
    })

})