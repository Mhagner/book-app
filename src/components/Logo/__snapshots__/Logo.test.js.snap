// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component Componente should render logo dark 1`] = `
<div
  class="logo__container"
  data-testid="logo"
>
  <img
    alt="ioasys"
    class="logo__image"
    src="logo-dark.png"
  />
  <span
    class="logo__text "
  >
    Books
  </span>
</div>
`;

exports[`Logo component Componente should render logo light default 1`] = `
<div
  class="logo__container"
  data-testid="logo"
>
  <img
    alt="ioasys"
    class="logo__image"
    src="logo.png"
  />
  <span
    class="logo__text  logo__text--light"
  >
    Books
  </span>
</div>
`;
